### ISSUE TITLE
Box not found in cxx::bridge scope when using alloc

### ISSUE THREAD

**Issue by crzysdrs** [dtolnay/cxx#1065]
`cxx::bridge` appears to be using `Box` instead of a fully elaborated name when `Box` is used in a function declaration, which prevents it from being found. There doesn't appear to be a way to import another path for `Box` in a `cxx::bridge` module (`use items are not allowed within cxx bridge`).

```
error[E0412]: cannot find type `Box` in this scope
  --> src/lib.rs:15:21
   |
15 |         fn foo() -> Box<Bar>;
   |                     ^^^ not found in this scope
   |
help: consider importing one of these items
   |
14 |         type use alloc::prelude::v1::Box;
   |              ++++++++++++++++++++++++++++
14 |         type use crate::Box;
   |              +++++++++++++++
   ```
   
```toml
# Cargo.toml
[package]
name = "cxx_test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
cxx = {version="1.0.71", default-features=false, features=["alloc"]}
```

```rust
// src/lib.rs
#![no_std]
extern crate alloc;
use alloc::boxed::Box;

struct Bar {}

fn foo() -> Box<Bar> {
    Box::new(Bar {})
}

#[cxx::bridge]
mod ffi {
    extern "Rust" {
        type Bar;
        fn foo() -> Box<Bar>;
    }
}
```

**Comment by nathanprat** [comment]
Hi,

I am pretty sure I am hitting this same issue, but with `Vec` and `String` instead of `Box`.

Did you find a way to make it work?

**Comment by dtolnay** [comment]
I published a fix in cxx 1.0.80. Please report any other cases you hit with this kind of error, this stuff is tricky to get right.
